src/Reference/HasOneSql.php

```diff
@@ -71,7 +71,7 @@ function (Model $ourModel) use ($theirFieldName) {
                 $theirModel = $this->getTheirModel();
 
                 $theirModel->addCondition($theirFieldName, $ourModel->get($ourFieldName));
-                $ourModel->set($this->getOurFieldName(), $theirModel->action('field', [$theirModel->id_field]));
+                $ourModel->set($this->getOurFieldName(), $theirModel->action('field', [$theirModel->id_field])->getOne());
```

Comment:
@DarkSide666 @georgehristov a bug?

Reply:
It is not a bug as it includes the SQL statement

Reply:
`$this->getOurFieldName()` will always point to a `FieldSqlExpression` Field?

if yes, `FieldSql` should still be RO (read only) and it seems it does not/should not support set at all